docs/guides-buidler-migration.md

```diff
@@ -0,0 +1,193 @@
+
+## 2. Add/replace npm scripts
+
+Since the app won't use `aragonCLI` anymore, some NPM scripts need to be changed.
+
+In `package.json`:
+
+  - "postinstall": "npm run build-app"
+  - "build-app": "cd app && npm install && cd .."
+  - "compile": "buidler compile --force"
+  - "start": "buidler start"
+  - "test": "buidler test --network buidlerevm"
+
+Also, make sure that you have the following two scripts in `app/package.json`:
+
+  - **serve**: Launches the web server (Previously called `devserver` script in most projects)
+  - **watch**: Watches for file changes (Previously called `watch:script` script in most projects)
+
+
+## 3. Update `.gitignore` file
+
+Add the following two folders to your `.gitignore` file:
+
+  - artifacts
+  - cache
+
+## 4. Create the Buidler config file
+
+Add the following `buidler.config.js` file to the root of your project: 
+
+```js
+const { usePlugin } = require('@nomiclabs/buidler/config')
+
+usePlugin('@aragon/buidler-aragon')
+
+module.exports = {
+  defaultNetwork: 'localhost',
+  networks: {
+    localhost: {
+      url: 'http://localhost:8545'
+    },
+  },
+  solc: {
+    version: '0.4.24',
+    optimizer: {
+      enabled: true,
+      runs: 10000
+    }
+  },
+  aragon: {
+    appServePort: 1234,
+    clientServePort: 3000,
+    appSrcPath: 'app/',
+    appBuildOutputPath: 'dist/',
+    hooks: require('./scripts/buidler-hooks')
+  }
+}
+```
+
+You can find more information about the Buidler configuration file [here](https://buidler.dev/config). As for the `aragon` section, it contains 5 main options:
+
+  - `appServePort`: The app server port started by the `serve` script of the `app` folder. (Default: `1234`)
+  - `clientServePort`: Aragon client server port.
+  - `appSrcPath`: Frontend source for the app.
+  - `appBuildOutputPath`: Built app path.
+  - `hooks`: Aragon hooks functions. (See [Hooks](#5-hooks) section for more information)
+
+## 5. Hooks
+
+Hooks are custom functions called at various stages of an app initialization. They can be used to execute any logic before or after the app is created and also to specify the app's [`initialize()`](https://hack.aragon.org/docs/aragonos-building#constructor-and-initialization) function parameters. For example, initializing the app with a `uint` and a `string` would be as simple as writing the following hook:
+
+```js
+  getInitParams: async ({}, { web3, artifacts }) => {
+    return [23, "Hello, World!"]
+  }
+```
+
+Hooks can also be used to replace Aragon's templates logic. You can find complete examples in 1Hive's [Conviction Voting](https://github.com/aragonone/conviction-voting-app/blob/buidlerized/buidler-app/scripts/buidler-hooks.js) and [Payroll](https://github.com/1Hive/payroll-app/blob/buidler-setup/buidler-app/scripts/buidler-hooks.js) apps. In our case, we will simply add empty functions since we don't need any initialization logic. So we will create a file named `buidler-hooks.js` in the `scripts` folder and add the following code:
```

Comment:
> **If you were using a template contract before to initialize organization state during `aragon run`, you should convert the smart contract logic into the hooks. Because hooks are much easier to write, maintain, and debug, the Buidler plugin will not be supporting template contracts.** ~Hooks can also be used to replace Aragon's templates logic.~